RFCs/FS-1100-Printf-binary.md

```diff
@@ -0,0 +1,85 @@
+# F# RFC FS-1100 - Printf format as binary number
+
+The design suggestion [Printf format as binary number](https://github.com/fsharp/fslang-suggestions/issues/1008) has been marked "approved in principle".
+
+This RFC covers the detailed proposal for this suggestion.
+
+- [x] Approved in principle [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/1008)
+- [ ] [Implementation](https://github.com/dotnet/fsharp/pull/FILL-ME-IN)
+- [ ] Design Review Meeting(s) with @dsyme and others invitees
+- [Discussion](https://github.com/fsharp/fslang-design/discussions/568)
+
+# Summary
+
+An additional format specifier `%B` will be added to F# which formats a basic integer type
+(currently (byte|int16|int32|int64|sbyte|uint16|uint32|uint64|nativeint|unativeint))
+as an unsigned binary number.
+
+# Motivation
+
+There are 4 number bases that are supported in `Convert.ToString`: 2, 8, 10 and 16.
+Of these, only base 2 does not have a corresponding printf format, as 8 has `%o`,
+10 has `%d` and `%i`, and 16 has `%x` and `%X`.
+
+Moreover, when introducing newcomers to programming, it is important to explain the
+binary number system before octal and hexadecimal, and this printf format provides
+a convenient way to view the representation of numbers in binary form before being
+introduced to octal and hexadecimal forms. This also avoids `Convert.ToString` boilerplate
+which gets in the way.
+
+As a result, not only is a consistent experience offered across the standard number bases,
+it also offers convenience when learning F#.
+
+# Detailed design
+
+This should be implemented by referencing the existing implementation of `%o`, along with
+its limitations (no signed numbers) to be consistent and be within expectations.
+
+```fs
+printf "%o" 123
+printf "%B" 123
+```
+```
+173
+1111011
+```
```

Comment:
```suggestion
The following F# code:

```fs
printf "%o" 123
printf "%B" 123
```

Prints:

```
173
1111011
```
```

Reply:
Wait... This messes up the formatting.